src/constants_manager.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#ifndef CONSTANTS_MANAGER_H
+#define CONSTANTS_MANAGER_H
+
+#include "json.h"
+
+#include <set>
+
+/** Constansts manager. To read constant use something like:
+float val = get_constant<float>( "FLOAT_CONSTANT_EXAMPLE" );
+get_constant() avalible statically*/
+class constants_manager
+{
+        friend class DynamicDataLoader;
+
+
+    private:
+        /** Load constants from JSON */
+        static void load( JsonObject &jo );
+
+    public:
+        class game_constant
+        {
+            public:
+                template<typename T>
+                T value_as() const;
+                game_constant( const std::string &id = std::string() ) : id_( id ) {}
+
+                /** Load constant definition from JSON */
+                void load( JsonObject &jo );
+
+            private:
+                const std::string id_;
+                std::string description_;
```

Comment:
What does the description? It seems unused. Is it really needed in the code?

Reply:
It may be useful for observation and analysis. And somewhat force to explain what is it and why do we need it.

Reply:
Comments can do it too. But comments can't be observed in game . Also modder can write info here about why do they overwrite that constant